### PR TITLE
Fix invalid index error on node disconnection.

### PR DESCRIPTION
**Pull request by sebaraba**
### What does this PR do?
When the latest `BlockNumber` was 0 `Invalid index error` would be fired because we try to `CatchUpToLatestAndSubscribeToNewBlocksObservable` starting from `index = -1`. 
By deleting the `.subtract(BigInt.ONE)` the block number will never be 0 and nodes would restart correctly.

### Where should the reviewer start?
*required*

### Why is it needed?
*required*

### CODE REVIEW

File: core/src/main/java/org/web3j/protocol/core/DefaultBlockParameter.java [hunk 1]
```diff
@@ -10,6 +10,8 @@
  */
 public interface DefaultBlockParameter {
     static DefaultBlockParameter valueOf(BigInteger blockNumber) {
+        if (BigInteger.ZERO.compareTo(blockNumber) >= 0)
+            blockNumber = BigInteger.ZERO;
```
Comment (iikirilov) [inline core/src/main/java/org/web3j/protocol/core/DefaultBlockParameter.java]: Here you should throw an appropriate exception rather than modifying the input.

File: integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java [hunk 2]
```diff
@@ -21,8 +21,8 @@
  */
 public class ObservableIT {
 
-    private static final int EVENT_COUNT = 5;
-    private static final int TIMEOUT_MINUTES = 5;
+    private static final int EVENT_COUNT = 1;
```
Comment (iikirilov) [inline integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java]: You should leave these values as they were.
You can add the extra static variable for edge cases like the one you are testing. i.e.
`private static final int SINGLE_EVENT_COUNT = 1`
Then add specific tests for those edge cases.

File: integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java [hunk 3]
```diff
@@ -21,8 +21,8 @@
...
+    private static final int TIMEOUT_MINUTES = 1;
```
Comment (iikirilov) [inline integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java]: Timeout can stay the same

File: integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java [hunk 4]
```diff
@@ -64,7 +64,7 @@ public void testCatchUpToLatestAndSubscribeToNewBlocksObservable() throws Except
                 .send();
         BigInteger latestBlockNumber = ethBlock.getBlock().getNumber();
         run(web3j.catchUpToLatestAndSubscribeToNewBlocksObservable(
-                new DefaultBlockParameterNumber(latestBlockNumber.subtract(BigInteger.ONE)),
+                new DefaultBlockParameterNumber(latestBlockNumber),
```
Comment (iikirilov) [inline integration-tests/src/test/java/org/web3j/protocol/core/ObservableIT.java]: Leave this test as it was and add another for your edge case.

File: core/src/main/java/org/web3j/protocol/core/DefaultBlockParameter.java (hunk 1 above)
Comment (conor10) [reply core/src/main/java/org/web3j/protocol/core/DefaultBlockParameter.java]: I'm not sure that this did fix the issue - @xaviarias did you have any feedback on it?

**Comment by xaviarias** [comment]
This could be merged and closed after implementing the requested changes.

**Comment by iikirilov** [comment]
There are also checkstyle violations.

**Comment by sebaraba** [comment]
@iikirilov  Done with this.
